src/pipx/commands/run.py

```diff
@@ -145,21 +189,17 @@ def _download_and_run(
         is_main_package=True,
     )
 
-    if not (venv.bin_path / app).exists():
-        apps = venv.pipx_metadata.main_package.apps
-        raise PipxError(
-            f"""
-            '{app}' executable script not found in package '{package_or_url}'.
-            Available executable scripts: {', '.join(b for b in apps)}
-            """
-        )
+    command = _resolve_command(venv, package_or_url, app)
 
     if not use_cache:
         # Let future _remove_all_expired_venvs know to remove this
         (venv_dir / VENV_EXPIRED_FILENAME).touch()
 
-    # This never returns
-    venv.run_app(app, app_args)
+    # These never return
+    if isinstance(command, EntryPoint):
+        venv.run_entry_point(command, app_args)
+    else:
+        venv.run_app(app, app_args)
```

Comment:
```suggestion
        venv.run_app(command, app_args)
```

gets changed on windows. Wish single dispatch method was available pre 3.8.

Reply:
It just occurs to me `f"{app}.exe"` is not actually needed. `subprocess` on Windows automatically resolves the extension (sine it is implemented with the `CreateProcess` Windows API).

Reply:
> This never returns

Why are these not annotated NoReturn instead of None, by the way? Would need fewer comments and the type checker would check the validity of the no return status. Unrelated, just thought I'd ask here.

Reply:
Isn't the windows test failing because of this? Or is it just tripping a unit test?

Reply:
> Why are these not annotated NoReturn instead of None, by the way?

Probably some legacy reasons, don’t know.

> Isn't the windows test failing because of this? Or is it just tripping a unit test?

The test fails because it cannot find a line in the logs; cowsay itself seems to run fine.

Reply:
Pretty sure I wrote the `None` return values because I was not aware there was a `NoReturn`.  Feel free to fix that.